vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -710,25 +681,17 @@ private void convertMongoIterable(MongoIterable iterable, Handler<AsyncResult<Js
 
   }
 
-  private DistinctIterable findDistinctValuesWithQuery(String collection, String fieldName, String resultClassname, JsonObject query, Handler resultHandler) {
+  private DistinctIterable<?> findDistinctValuesWithQuery(String collection, String fieldName, String resultClassname, JsonObject query) throws ClassNotFoundException {
     requireNonNull(collection, "collection cannot be null");
     requireNonNull(fieldName, "fieldName cannot be null");
-    requireNonNull(resultHandler, "resultHandler cannot be null");
     requireNonNull(query, "query cannot be null");
 
     JsonObject encodedQuery = encodeKeyWhenUseObjectId(query);
 
     Bson bquery = wrap(encodedQuery);
 
-    final Class resultClass;
-    try {
-      resultClass = Class.forName(resultClassname);
-    } catch (ClassNotFoundException e) {
-      resultHandler.handle(Future.failedFuture(e));
-      return null;
-    }
     MongoCollection<JsonObject> mongoCollection = getCollection(collection);
-    return mongoCollection.distinct(fieldName, bquery, resultClass);
+    return mongoCollection.distinct(fieldName, bquery, Class.forName(resultClassname));
```

Comment:
Don't use Class.forName, but this.getClass().getClassloader().load(resultClassname);

Class.forName hijacks the classloader hierarchy to delegate to the application classloader (classpath)

Reply:
I didn't want to change the original code but, fine, I'll do it.